receiver/splunkhecreceiver/receiver.go

```diff
@@ -253,6 +330,17 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidEncodingRespBody, 0, errInvalidEncoding)
 		return
 	}
+	var partitionID string
+	var partitionErr error
+	if partitionID, partitionErr = r.validateAndExtractChannelHeader(req); partitionErr != nil {
+		r.failRequest(ctx, resp, http.StatusBadRequest, []byte(partitionErr.Error()), 0, partitionErr)
+		return
+	}
+
+	if len(partitionID) > 0 && r.ackExt == nil {
+		// when channel id is provided in request but ackExt is not set up, fail the request.
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, 0, nil)
+	}
```

Comment:
this will break existing hec receiver users that don't have ack extension set up. I think for backward compatibility sake we should ignore the partition ID silently

Reply:
addressed